gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -325,7 +328,10 @@ private ByteString readRequestData() throws IOException {
             });
       }
 
-      public Object getResponse() {
+      public Object getResponse() throws IOException {
+	if (hasError()) {
```

Comment:
I think instead of/in addition to doing this here, you also want to add this check above at while (!responseObserver.isFinished()), to not let while loop stop upon error. Instead it should only stop if and only if it finishes without error or maxmimum retry reached in which case exception is thrown.

Reply:
Ah, good point. I'll have it check only whether the upload was finalized instead.